### ISSUE TITLE
FRE: New tab "tab button"

### ISSUE THREAD

**Issue by MarcinOrlowski** [cbucher/console#432]
I'd love to have "+" tab as we have in browsers that would open new tab:

![newtab](https://cloud.githubusercontent.com/assets/8041294/24964558/ed8e91e8-1fa1-11e7-9e3a-f865468ca00f.png)

It'd be perfectly fine to have it act as `Edit -> Clone in New Tab` or from predefined tabs.

**Comment by cbucher** [comment]
Hi

- act as `Edit -> Clone in New Tab`
In browsers "+" means add a new tab. If we mimic the visual style, we must mimic the behavior too.

- act as `File -> New Tab`
You can already open a new tab by clicking on free tab area with middle mouse button.
Is the middle mouse button click satisfactory? Or is "+" visual indication essential?

Regards

**Comment by MarcinOrlowski** [comment]
Hi,

> act as Edit -> Clone in New Tab
> In browsers "+" means add a new tab. If we mimic the visual style, we must mimic the behavior too.

Well, not really. In browser empty tab makes sense (and also is technically doable) while here you cannot simply have "empty terminal" :)

> You can already open a new tab by clicking on free tab area with middle mouse button.
Is the middle mouse button click satisfactory?

If that'd help having this implemented then for me "+" can just work like middle click you mentioned. The middle click is bad IMHO for at least two reasons: 1) it's pretty much non-intuitive - I am using your console for long time and never thought that I'd try MMB there, 2) as most users nowadays MMB is roll. I hate when I need to clicking that roll :)

> Or is "+" visual indication essential?

"+" is simply compact, does not really need much translations :) and it simply tells (IMHO again) what it is for and what to expect when clicked. Chrome uses just something like this:

![chrome](https://cloud.githubusercontent.com/assets/8041294/25049428/156da036-2142-11e7-81c2-ceeffcba8fdf.png)

but I personally like "+" more as it (somehow) describes the outcome/functionality. For example this is what Firefox uses (unfocused/focused):

![ff](https://cloud.githubusercontent.com/assets/8041294/25049681/54f71664-2143-11e7-94f0-9001283abe72.png) ![ff-focus](https://cloud.githubusercontent.com/assets/8041294/25049684/5774283c-2143-11e7-941e-ed5a8c31fdc3.png)


and my visualisation is pretty much mixed approach in terms of appearance.

**Comment by cbucher** [comment]
> Well, not really. In browser empty tab makes sense (and also is technically doable) while here you cannot simply have "empty terminal" :)

Simply, "+" means "add a new thing" not "clone a thing".

> 1) it's pretty much non-intuitive - I am using your console for long time and never thought that I'd try MMB there

This is the behavior of Firefox.

> 2) as most users nowadays MMB is roll.

Here, I disagree. MMB is commonly used in browsers.
MMB click on a tab will close it.
MMB click on a link will open it in a new tab = (CTRL+LMB).

> "+" is simply compact, does not really need much translations :)

Yes, but the option to disable it will need translations.
Tooltip could reuse existing command text.

**Comment by cbucher** [comment]
I had the same idea some years ago. I set this idea aside because it costs development time for something redundant. It is basically a button in tabs strip. You already have this button in toolbar. You can also use keyboard shortcut, mouse shortcut and menu.

I propose to create a poll.

# People, who are interested by this feature, can comment here.

**Comment by cbucher** [comment]
@MarcinOrlowski Please, try last experimental version.

https://github.com/cbucher/console/wiki/Downloads

**Comment by artie-finkelstein** [comment]
I like the new tab button [+].  I don't keep the toolbar open, but miss the 'twist down' menu on the new window button; I still don't find the MMB usage to be intuitive, and using the the File/New Tab expansion seems clunky to me, but I put up with it.  Coupled with the 'allow closing/detaching the last tab/view' setting, the new tab button make it very easy to kill a session and open a new instance.  I guess the 'web browser' interface paradigm is taking over.  <grin>  Thank you for the 'redundant' work, I think it's a good addition.

**Comment by cbucher** [comment]
@artie-finkelstein thx

**Comment by artie-finkelstein** [comment]
I just installed the "1.18.2.17242 x64 Release Legacy" version and saw that the new 'new tab button' [+] was missing.  So I installed the "1.18.2.17242 x64 Release"version and it was back.  I'm not a big fan nor user of the MS Aero interface, but from what little I know, this isn't a UI feature I would associate with the over exuberant alpha blending that seems to be the Aero stock in trade.  Perhaps this feature linkage was intentional?  For know, I'll stay with the non-legacy version.  FWIW, my installation approach is to make copy of the current ConsoleZ directory, rename it show the soon to be previous version and then just copy the new version release files.  Diff'ing the 'diagnose' tmp files only shows differences for the first line (aero .vs. legacy) and the path to the settings file (since they are in different locations, but the settings files themselves are duplicates). 

I just did a quick regression test, and found the same lack of the 'new tab' tab on the "1.18.2.17172 x64 Release Legacy" dev version.  I may not use or rely on Aero, but I've obviously been using the Aero version of ConsoleZ.
